src/cargo/core/compiler/job_queue/mod.rs

```diff
@@ -952,7 +952,9 @@ impl<'cfg> DrainState<'cfg> {
                 }
 
                 for warning in output.warnings.iter() {
-                    bcx.config.shell().warn(warning)?;
+                    let package = format!("[{} {}]", unit.pkg.name(), unit.pkg.version());
+                    let warning_with_package = format!("{} {}", package, warning);
```

Comment:
I think we can merge them together to get a better reading experience.

Reply:
done @hi-rustin